DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,172 @@
+            int day = groups["day"].Success ? int.Parse(groups["day"].Value) : 0;
+            int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
+            int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
+            int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
+            int millisecond = groups["millisecond"].Success ? int.Parse(groups["millisecond"].Value) : 0;
+
+            if (groups["timeZone"].Success)
+            {
+                int timeZoneHour = int.Parse(groups["timeZoneHour"].Value) * int.Parse(groups["sign"].Value + "1");
+                int timeZoneMinute = int.Parse(groups["timeZoneMinute"].Value);
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond > 999 ? int.Parse(millisecond.ToString().Substring(0, 3)) : millisecond, new TimeSpan(timeZoneHour, timeZoneMinute, 0)),
+                    HasTimeZone = true,
+                };
+            }
+            else
+            {
+                return new DateTimeObject()
+                {
+                    DateValue = new DateTimeOffset(year, month, day, hour, minute, second, millisecond, default),
+                    HasTimeZone = false,
+                };
+            }
+        }
+
+        public static string GenerateDicomDateString(DateTimeOffset date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            return date.ToString("yyyyMMdd", CultureInfo.InvariantCulture);
+        }
+
+        public static string GenerateDicomDateTimeString(DateTimeObject date)
+        {
+            if (date == null)
+            {
+                return null;
+            }
+
+            if (date.HasTimeZone == null || !(bool)date.HasTimeZone)
+            {
+                return date.DateValue.ToString("yyyyMMddhhmmss.ffffff", CultureInfo.InvariantCulture);
+            }
+            else
+            {
+                return date.DateValue.ToString("yyyyMMddHHmmss.ffffffzzz", CultureInfo.InvariantCulture).Replace(":", string.Empty);
+            }
+        }
+
+        public static AgeValue ParseAge(string age)
+        {
+            if (string.IsNullOrEmpty(age))
+            {
+                return null;
+            }
+
+            Dictionary<string, AgeType> ageTypeMapping = new Dictionary<string, AgeType>
+            {
+                { "Y", AgeType.Year },
+                { "M", AgeType.Month },
+                { "W", AgeType.Week },
+                { "D", AgeType.Day },
+            };
+
+            foreach (var item in ageTypeMapping)
+            {
+                if (new Regex(@"\d{3}" + item.Key).IsMatch(age))
+                {
+                    return new AgeValue(uint.Parse(age.Substring(0, 3)), item.Value);
```

Comment:
>3 [](http://example.com/codeflow?start=68&length=1)

how about creating a variable instead of magic number? I found that there are other places too, so please change themin the same way.

Reply:
Set 3 as a constants variable "AgeStringLength"